compiler/qsc_linter/src/lints.rs

```diff
@@ -9,9 +9,20 @@ macro_rules! lint {
         Lint {
             span: $span,
             level: $lint.level,
-            message: $lint.message,
-            help: $lint.help,
-            kind: $lint.kind,
+            message: $lint.message(),
+            help: $lint.help(),
+            kind: $lint.lint_kind(),
+            code_action_edits: vec![],
```

Comment:
I can kind of see why this is tricky to achieve with the current macros, but it'd be great if we made `code_action_edits()` into a method so that we can lazily compute the edits. I doubt we need to worry about perf any time soon though.

Reply:
The way an edit is constructed can vary wildly from one lint to another, and also require all kinds of various information. I'm not seeing how to make a single function type for this field that doesn't result in doing all the heavy lifting at the linter still anyways and just having a getter function here that takes no args. At that point the complexity isn't worth it.

Maybe you had something different in mind?